packages/server/server/api/user/user.service.ts

```diff
@@ -44,25 +44,35 @@ class UserService {
             });
     }
 
-    public checkDuplicateUser(userData: User): Promise<boolean> {
+    public checkDuplicateUser(userData: User): Promise<string | null> {
```

Comment:
Maybe you could do something like this for readability.
```
const duplicateEmail = userModel.findOne({ email: userData.email })
  .collation({ locale: 'en', strength: 2 })
  .exec();
const duplicateUsername = userModel.findOne({ username: userData.username })
  .collation({ locale: 'en', strength: 2 }).exec();

return Promise.all([duplicateEmail, duplicateUsername]).then(([byEmail, byUsername]: [User, User]) => {
    if (byEmail) {
        return Promise.resolve('Email');
    } else if (byUsername) {
        return Promise.resolve('Username');
    } else {
        return Promise.resolve(null);
    }
})
.catch((err) => {
     logger.warn(`Failed to find user`);
      return Promise.reject(
            new InternalServerError({
                message: `Failed to find user: ${err.toString()}`,
            })
       );
});
```

Reply:
Thank you!! I forgot `Promise.all` is a thing, and this is so much better!